macros/src/codegen.rs

```diff
@@ -390,11 +412,15 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
                             };
                             quote! {
                                 let _data = self.context.#a(#temporary_context_call #g_a_param) #action_await ;
+                                    self.context_mut().#exit_ident();
+                                    self.context_mut().#entry_ident();
                                 self.state = Some(#states_type_name::#out_state);
                             }
                         } else {
                             quote! {
                                 self.state = Some(#states_type_name::#out_state);
+                                    self.context_mut().#exit_ident();
+                                    self.context_mut().#entry_ident();
```

Comment:
I don't want to force everyone to implement these if they're not needed. Can we guard these calls based on their existence in the FSM definition?

Reply:
Yes, I have seen other functions to be behind these flags, will update this accordingly (and will put this MR in draft while I haven't done that).

When I have this flag and the other comments fixed, I think it is good to go.